contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -0,0 +1,639 @@
+        // by default use uni pool to swap as it has lower fees
+        useCRVPool = false;
+        // Initialize CRV pool to 3pool
+        crvPool = address(0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7);
+    }
+
+    event Cloned(address indexed clone);
+
+    /*
+     * @notice
+     *  Cloning function to migrate/ deploy to other pools
+     * @param _providerA, provider strategy of tokenA
+     * @param _providerB, provider strategy of tokenB
+     * @param _referenceToken, token to use as reference, for pricing oracles and paying hedging costs (if any)
+     * @param _pool, Uni V3 pool to LP
+     * @param _ticksFromCurrent, # of ticks up & down to provide liquidity into
+     * @return newJoint, address of newly deployed joint
+     */
+    function cloneUniV3StablesJoint(
+        address _providerA,
+        address _providerB,
+        address _referenceToken,
+        address _pool,
+        uint24 _ticksFromCurrent
+    ) external returns (address newJoint) {
+        require(isOriginal, "!original");
+        bytes20 addressBytes = bytes20(address(this));
+
+        assembly {
+            // EIP-1167 bytecode
+            let clone_code := mload(0x40)
+            mstore(
+                clone_code,
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newJoint := create(0, clone_code, 0x37)
+        }
+
+        UniV3StablesJoint(newJoint).initialize(
+            _providerA,
+            _providerB,
+            _referenceToken,
+            _pool,
+            _ticksFromCurrent
+        );
+
+        emit Cloned(newJoint);
+    }
+
+    /*
+     * @notice
+     *  Function returning the name of the joint in the format "NoHedgeUniV3StablesJoint(USDC-DAI)"
+     * @return name of the strategy
+     */
+    function name() external view override returns (string memory) {
+        string memory ab = string(
+            abi.encodePacked(
+                IERC20Extended(address(tokenA)).symbol(),
+                "-",
+                IERC20Extended(address(tokenB)).symbol()
+            )
+        );
+
+        return string(abi.encodePacked("NoHedgeUniV3StablesJoint(", ab, ")"));
+    }
+
+    /*
+     * @notice
+     *  Function returning the liquidity amount of the LP position
+     * @return liquidity from positionInfo
+     */
+    function balanceOfPool() public view override returns (uint256) {
+        IUniswapV3Pool.PositionInfo memory positionInfo = _positionInfo();
+        return positionInfo.liquidity;
+    }
+
+    /*
+     * @notice
+     *  Function available for vault managers to set the CRV pool to use for swaps
+     * @param newPool, new CRV pool address to use
+     */
+    function setCRVPool(address newPool) external onlyVaultManagers {
```

Comment:
this will trigger an alarm during security process as VaultMAnagers could inject a new mailicious contract here.

Reply:
Changing it to `onlyGovernance`